azurerm_private_dns_zone/main.tf

```diff
@@ -0,0 +1,18 @@
+provider "azurerm" {
+  version = "=2.4.0"
+  features {}
+}
+
+terraform {
+  backend "azurerm" {}
+}
+
+resource "azurerm_private_dns_zone" "private_dns_zone" {
+  count               = var.name == null ? 0 : 1
```

Comment:
I wand to have this module optional since I can create it within the private link

Reply:
We can use the same rule of the other modules

https://github.com/pagopa/io-infrastructure-modules-new/blob/master/azurerm_subnet/vars.tf#L50